AzureMonitorAgent/agent.py

```diff
@@ -573,23 +573,24 @@ def enable():
         hutil_log_info("This VM is an Arc VM, Running the arc watcher daemon.")
         start_arc_process()
 
-    if is_systemd():
-        AMAServiceStartCommand = "systemctl start azuremonitoragent"
-        AMAServiceStatusCommand = "systemctl status azuremonitoragent"
-    else:
-        hutil_log_info("The VM doesn't have systemctl. Using the init.d service to start azuremonitoragent.")
-        AMAServiceStartCommand = "/etc/init.d/azuremonitoragent start"
-        AMAServiceStatusCommand = "/etc/init.d/azuremonitoragent status"
-
     public_settings, protected_settings = get_settings()
 
     if public_settings is not None and public_settings.get("GCS_AUTO_CONFIG") == "true":
-        AMAServiceStartCommand = "systemctl start azuremonitoragentmgr"
+        # The systemd service azuremonitoragentmgr is disabled by default.
+        # Enable it here so the service starts after reboot.
+        AMAServiceStartCommand = "systemctl start azuremonitoragentmgr && systemctl enable azuremonitoragentmgr"
```

Comment:
nit; could we simplify the code here a bit by parameterizing these commands, e.g.
```
if public_settings is not None and public_settings.get("GCS_AUTO_CONFIG") == "true":
    service_name = "azuremonitoragentmgr"
else:
    service_name = "azuremonitoragent"

...

AMAServiceStartCommand = "systemctl start {0} && systemctl enable {0}".format(service_name)
```

Reply:
Addressed in latest update